src/common/menu.js

```diff
@@ -37,19 +37,35 @@ const menuData = [
   },
 ];
 
-function formatter(data, parentPath = '', parentAuthority) {
+// Generates a configuration of type { Path: Name }
+// from menuData. A level is used to distinguish
+// root paths from other paths.
+function formatter(data, parentPath = '', parentName = '', level = 1) {
   return data.map(item => {
-    let { path } = item;
+    let { path, name } = item;
     if (!isUrl(path)) {
       path = parentPath + item.path;
     }
-    const result = {
-      ...item,
-      path,
-      authority: item.authority || parentAuthority,
-    };
+    // For all non root paths,
+    // append the parent name with
+    // current name.
+    if (level !== 1) {
+      name = `${parentName}/${name}`;
+    }
+    let result = {};
     if (item.routes) {
-      result.routes = formatter(item.routes, `${parentPath}${item.path}`, item.authority);
+      result = {
```

Comment:
We can avoid duplicate assignment to `result` in the if else block by moving it outside of the block (DRY principle).